packages/x-charts/src/internals/plugins/corePlugins/useChartInteractionListener/useChartInteractionListener.ts

```diff
@@ -0,0 +1,154 @@
+  hoverAction,
+]);
+
+export const useChartInteractionListener: ChartPlugin<UseChartInteractionListenerSignature> = ({
+  svgRef,
+}) => {
+  const listenersRef = React.useRef<ListenerRef>(new Map());
+
+  const retriggerEvent = React.useCallback((interaction: ChartInteraction, state: any) => {
+    const listeners = listenersRef.current.get(interaction);
+    const memo = !state.memo ? new Map<Function, any>() : state.memo;
+
+    if (listeners) {
+      listeners.forEach((callback) => {
+        const result = callback({ ...state, memo: memo.get(callback) });
+        if (result) {
+          memo.set(callback, result);
+        }
+      });
+    }
+
+    return memo;
+  }, []);
+
+  useGesture(
+    {
+      onDrag: (state) => retriggerEvent('drag', state),
+      onDragStart: (state) => retriggerEvent('dragStart', state),
+      onDragEnd: (state) => retriggerEvent('dragEnd', state),
+      onPinch: (state) => retriggerEvent('pinch', state),
+      onPinchStart: (state) => retriggerEvent('pinchStart', state),
+      onPinchEnd: (state) => retriggerEvent('pinchEnd', state),
+      onWheel: (state) => retriggerEvent('wheel', state),
+      onWheelStart: (state) => retriggerEvent('wheelStart', state),
+      onWheelEnd: (state) => retriggerEvent('wheelEnd', state),
+      onMove: (state) => retriggerEvent('move', state),
+      onMoveStart: (state) => retriggerEvent('moveStart', state),
+      onMoveEnd: (state) => retriggerEvent('moveEnd', state),
+      onHover: (state) => retriggerEvent('hover', state),
+    },
+    {
+      target: svgRef,
+      eventOptions: {
+        passive: false,
+      },
+      drag: {
+        pointer: {
+          // We can allow customizing the number of pointers
+          buttons: 1,
+          keys: false,
+          // Disable using `setPointerCapture` when testing, as it doesn't work properly.
+          capture: process.env.NODE_ENV !== 'test',
+        },
+      },
+      pinch: {},
+    },
+  );
+
+  const addInteractionListener: AddInteractionListener = React.useCallback(
+    (interaction, callback) => {
+      let listeners = listenersRef.current.get(interaction);
+
+      if (!listeners) {
+        listeners = new Set<ChartInteractionHandler<any, any>>();
+        listeners.add(callback);
+        listenersRef.current.set(interaction, listeners);
+      } else {
+        listeners.add(callback);
+      }
+
+      return {
+        cleanup: () => listeners.delete(callback),
+      };
+    },
+    [],
+  );
+
+  const addMultipleInteractionListeners: AddMultipleInteractionListeners = React.useCallback(
+    (interactions, callback) => {
+      const cleanups = interactions.map((interaction) =>
+        // @ts-expect-error Overriding the type because the type of the callback is not inferred
+        addInteractionListener(interaction, callback),
+      );
+      return {
+        cleanup: () => cleanups.forEach((cleanup) => cleanup.cleanup()),
+      };
+    },
+    [addInteractionListener],
+  );
```

Comment:
Not sure if this would work, but probably worth a shot:

```suggestion
  const addMultipleInteractionListeners = React.useCallback<AddMultipleInteractionListeners>(
    (interactions, callback) => {
      const cleanups = interactions.map((interaction) =>
        addInteractionListener(interaction, callback),
      );
      return {
        cleanup: () => cleanups.forEach((cleanup) => cleanup.cleanup()),
      };
    },
    [addInteractionListener],
  );
```

Reply:
That resolves to the same type, so it doesn't work.